modules/porous_flow/include/kernels/PorFlowComponentMassTimeDerivative.h

```diff
@@ -0,0 +1,79 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
+#ifndef PORFLOWCOMPONENTMASSTIMEDERIVATIVE_H
+#define PORFLOWCOMPONENTMASSTIMEDERIVATIVE_H
+
+#include "TimeDerivative.h"
+#include "PorFlowVarNames.h"
+
+// Forward Declarations
+class PorFlowComponentMassTimeDerivative;
+
+template<>
+InputParameters validParams<PorFlowComponentMassTimeDerivative>();
+
+/**
+ * Kernel = (mass_component - mass_component_old)/dt
+ * where mass_component = porosity*sum_phases(density_phase*saturation_phase*massfrac_phase^component)
+ * It is lumped to the nodes
+ */
+class PorFlowComponentMassTimeDerivative : public TimeKernel
+{
+public:
+
+  PorFlowComponentMassTimeDerivative(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual();
+
```

Comment:
Using empty lines in this manner does not seem particularly useful to me. IMO they should be used sparingly to visually separate blocks that go together. In this particular file I'd suggest adding `///` comments to document the members as well.